src/legacy/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -121,37 +120,30 @@ uiRoutes
     resolve: {
       ip: function (Promise, indexPatterns, config, Private) {
         const State = Private(StateProvider);
-        const savedObjectsClient = Private(SavedObjectsClientProvider);
```

Comment:
dear @cjcenizal, have a question, asking you because, you introduced the indexpatterns service here. do you think it's good to use the saved objects cache indexpatterns service to prevent a 2nd saved object request to Elasticsearch here? A user created an issue, said we're making duplicate requests. I took a look, modified code, now there's only one request. Are there any pitfalls here?

Reply:
@kertal I don't remember creating the index patterns service. Did you see a PR somewhere in which I created it?

I may not be the best person to provide feedback on this PR because it's been so long since I've worked on Index Patterns or Discover, so I'm going to remove myself as reviewer.

However, just from looking at the code my biggest concern is the initial population of the cache. What guarantee do we have that the cache will be populated by the time this code runs?

Reply:
thx @cjcenizal was pinging you because you've introduced the `indexPatterns` a long time ago. The initial population I will check, than there should be no reason left not to remove one request